src/encoded/search_views.py

```diff
@@ -0,0 +1,252 @@
+from snovault.elasticsearch.searches.fields import IDResponseField
+from snovault.elasticsearch.searches.fields import NotificationResponseField
+from snovault.elasticsearch.searches.fields import NonSortableResponseField
+from snovault.elasticsearch.searches.fields import RawMatrixWithAggsResponseField
+from snovault.elasticsearch.searches.fields import RawSearchWithAggsResponseField
+from snovault.elasticsearch.searches.fields import SearchBaseResponseField
+from snovault.elasticsearch.searches.fields import SortResponseField
+from snovault.elasticsearch.searches.fields import TitleResponseField
+from snovault.elasticsearch.searches.fields import TypeOnlyClearFiltersResponseField
+from snovault.elasticsearch.searches.fields import TypeResponseField
+from snovault.elasticsearch.searches.parsers import ParamsParser
+from snovault.elasticsearch.searches.responses import FieldedResponse
+
+
+def includeme(config):
+    config.add_route('search', '/search{slash:/?}')
+    config.add_route('searchv2_raw', '/searchv2_raw{slash:/?}')
+    config.add_route('searchv2_quick', '/searchv2_quick{slash:/?}')
+    config.add_route('report', '/report{slash:/?}')
+    config.add_route('matrixv2_raw', '/matrixv2_raw{slash:/?}')
+    config.add_route('matrix', '/matrix{slash:/?}')
+    config.add_route('summary', '/summary{slash:/?}')
+    config.add_route('audit', '/audit{slash:/?}')
+    config.scan(__name__)
+
+
+DEFAULT_ITEM_TYPES = [
+    'AntibodyLot',
+    'Award',
+    'Biosample',
+    'BiosampleType',
+    'Dataset',
+    'GeneticModification',
+    'Page',
+    'Pipeline',
+    'Publication',
+    'Software',
+    'Gene',
+    'Target',
+    'File',
+    'Lab'
+]
+
+
+@view_config(route_name='search', request_method='GET', permission='search')
+def search(context, request):
+    # Note the order of rendering matters for some fields, e.g. AllResponseField and
+    # NotificationResponseField depend on results from BasicSearchWithFacetsResponseField.
+    fr = FieldedResponse(
+        _meta={
+            'params_parser': ParamsParser(request)
+        },
+        response_fields=[
+            TitleResponseField(
+                title=SEARCH_TITLE
+            ),
+            TypeResponseField(
+                at_type=[SEARCH_TITLE]
+            ),
+            IDResponseField(),
+            ContextResponseField(),
+            BasicSearchWithFacetsResponseField(
+                default_item_types=DEFAULT_ITEM_TYPES
+            ),
+            AllResponseField(),
+            NotificationResponseField(),
+            FiltersResponseField(),
+            ClearFiltersResponseField(),
+            ColumnsResponseField(),
+            SortResponseField(),
+            DebugQueryResponseField()
+        ]
+    )
+    return fr.render()
+
+
+@view_config(route_name='searchv2_raw', request_method='GET', permission='search')
+def searchv2_raw(context, request):
+    fr = FieldedResponse(
```

Comment:
(Minor) Would it be a good idea to place the declaration of things like _meta outside of the signature of FieldedResponse?